app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,67 @@
+class RecipesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_recipe, only: %i[show edit update destroy]
+
+  # GET /recipes or /recipes.json
+  def index
+    @current_user = current_user
+    @recipes = @current_user.recipes
+  end
+
+  # GET /recipes/1 or /recipes/1.json
+  def show
+    user = current_user
+    @recipe = user.recipes.find(params[:id])
+    @foods = RecipeFood.where(recipe_id: @recipe.id).includes(:food)
+  end
+
+  # GET /recipes/new
+  def new
+    @recipe = Recipe.new
+  end
+
+  # GET /recipes/1/edit
+  def edit; end
+
+  # POST /recipes or /recipes.json
+  def create
+    @recipe = Recipe.new(recipe_params)
+    @recipe.user = current_user
+    respond_to do |format|
+      if @recipe.save
+        format.html { redirect_to recipes_url, notice: 'Recipe was successfully created.' }
+        format.json { render :show, status: :created, location: @recipe }
+      else
+        flash[:error] = 'Error: recipe could not be saved'
+        redirect_to new_recipe_url
+      end
+    end
```

Comment:
- [ ] Creating a new recipe entry doesn't work as expected and shows an error which I have attached below 👇 Please try to fix this bug 🐛 because the ability to create recipes is essential to the complete testing of this application 🙏 

![image](https://user-images.githubusercontent.com/46839548/209442994-ad73a3e6-336e-4e7d-8623-870a9ebbac42.png)

Reply:
@codepantha we are stuck on these two methods "create" food & "create" recipe. Can you please kindly help us understand how to implement them. What are we doing wrong in our logic. We have really tried but we are stuck here. Thank you.